File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys

i_n = int(sys.stdin.readline())
dq_out = deque()
for i_1 in range(i_n):
    tup_in = tuple(sys.stdin.readline().split())
    if tup_in[0] == 'push_front':
        dq_out.appendleft(tup_in[1])
    elif tup_in[0] == 'push_back':
        dq_out.append(tup_in[1])
    elif tup_in[0] == 'pop_front':
        if dq_out:
            print(dq_out.popleft())
        else:
            print(-1)
    elif tup_in[0] == 'pop_back':
        if dq_out:
            print(dq_out.pop())
        else:
            print(-1)
    elif tup_in[0] == 'size':
        print(len(dq_out))
    elif tup_in[0] == 'empty':
        if dq_out:
            print(0)
        else:
            print(1)
    elif tup_in[0] == 'front':
        if dq_out:
            print(dq_out[0])
        else:
            print(-1)
    elif tup_in[0] == 'back':
        if dq_out:
            print(dq_out[-1])
        else:
            print(-1)

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>
using namespace std;

int main()
{
	int N;
	cin >> N;
	deque<int> dq;

	while (N--)
	{
		string command_in;
		cin >> command_in;
		if (!command_in.compare("push_front"))
		{
			int command_arg;
			cin >> command_arg;
			dq.push_front(command_arg);
		}
		if (!command_in.compare("push_back"))
		{
			int command_arg;
			cin >> command_arg;
			dq.push_back(command_arg);
		}
		if (!command_in.compare("pop_front"))
		{
			if (dq.empty())
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << dq.front() << "\n";
				dq.pop_front();
			}
		}
		if (!command_in.compare("pop_back"))
		{
			if (dq.empty())
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << dq.back() << "\n";
				dq.pop_back();
			}
		}
		if (!command_in.compare("size"))
		{
			cout << dq.size() << "\n";
		}
		if (!command_in.compare("empty"))
		{
			if (dq.empty())
			{
				cout << 1 << "\n";
			}
			else
			{
				cout << 0 << "\n";
			}
		}
		if (!command_in.compare("front"))
		{
			if (dq.empty())
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << dq.front() << "\n";
			}
		}
		if (!command_in.compare("back"))
		{
			if (dq.empty())
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << dq.back() << "\n";
			}
		}
	}
	return 0;
}